### PR TITLE
MicroUI Example

### PR DESCRIPTION
**Pull request by HailToDodongo**
This adds a new example for a immediate-mode UI micro-ui.

### CODE REVIEW

File: examples/micro-ui/lib/microui.c [hunk 1]
```diff
@@ -0,0 +1,1232 @@
+// dear imgui, v1.90 WIP
```
Comment (sp1187) [inline examples/micro-ui/lib/microui.c]: Why does it say it is dear imgui here?

Comment (HailToDodongo) [reply examples/micro-ui/lib/microui.c]: huh, that's not in the original repo.
i tested a few single-file libs before choosing microui by lazily pasting them in, gonna remove that

File: examples/micro-ui/micro-ui.c [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    dfs_init(DFS_DEFAULT_LOCATION);
+    display_init(RESOLUTION_320x240, DEPTH_16_BPP, 3, GAMMA_NONE, FILTERS_RESAMPLE_ANTIALIAS_DEDITHER);
+    rdpq_init();
+    gl_init();
+    
+    #ifdef DEBUG_RDP
+        rdpq_debug_start();
+    #endif
+
+    surface_t surface = surface_alloc(FMT_RGBA16, display_get_width(), display_get_height());
```
Comment (rasky) [inline examples/micro-ui/micro-ui.c]: Maybe call this zbuffer?

File: examples/micro-ui/micro-ui.c [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        
+        glCallList(dpl_cube);
+        glPopMatrix();
+    }
+}
+
+int main(void)
+{
+    debug_init_isviewer();
+	debug_init_usblog();
```
Comment (rasky) [inline examples/micro-ui/micro-ui.c]: Extraneous indentation

File: examples/micro-ui/micro-ui.c [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+        // Game renderer
+        rdpq_attach(display_get(), &surface);
+
+        gl_context_begin();
+        glClearColor(clear_color[0], clear_color[1], clear_color[2], 1);
+        glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+
+        game_draw(); // (your games renderer)
+
+        mu64_draw(); // Step 5/5: render out the UI at the very end
```
Comment (rasky) [inline examples/micro-ui/micro-ui.c:297]: Given that this uses rdpq rather than OpenGL, shouldn't it go after `gl_context_end()`?

**Review by rasky — COMMENTED** [review]
Thank you very much! This GUI example is really great

**Comment by HailToDodongo** [comment]
issues are fixed, and all examples i wanted to showcase are added.

**Pull request merged**